docs/source/extending.rst

```diff
@@ -0,0 +1,144 @@
+.. _extending:
+
+======================
+Extending the Notebook
+======================
+
+Certain subsystems of the notebook server are designed to be extended or
+overridden by users.  This document explains these systems and shows how to
+override the notebook's defaults with your own custom behavior.
+
+Contents API
+------------
+
+.. currentmodule:: notebook.services.contents
+
+The Jupyter Notebook web application provides a graphical interface for
+creating, opening, renaming, and deleting files in a virtual filesystem.
+
+The :class:`ContentsManager<manager.ContentsManager>` class defines an abstract
```

Comment:
I think it should be possible to hide all but the last part of the name by writing these like this:

```
:class:`~manager.ContentsManager`
```